extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java

```diff
@@ -63,7 +65,28 @@ public void validate(ValidationContext validationContext) {
                             // No need to validate properties with default values
                             continue;
                         }
-                        String propertyName = configProperty.value("name").asString();
+                        AnnotationValue nameValue = configProperty.value("name");
+                        String propertyName;
+                        if (nameValue != null) {
+                            propertyName = nameValue.asString();
+                        } else {
+                            // org.acme.Foo.config
```

Comment:
It would be nice to have a test.
If, for some reason, the spec decides to change how default names are deduced, we won't be able to detect the need for change here.

Reply:
There is a TCK test for this and since we aim to run the TCK soon I thought another test would be redundant?

Reply:
Ok, in that case, it should be fine